.github/workflows/ci.yml

```diff
@@ -17,16 +17,89 @@ jobs:
         with:
           node-version: ${{ matrix.node }}
           cache: 'npm'
-      - run: npm ci
+      - uses: actions/cache@v3
+        id: npm-cache
+        with:
+          path: |
+            node_modules
+          key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-npm
+      - name: Install dependencies if cache invalid
+        if: steps.npm-cache.outputs.cache-hit != 'true'
+        run: npm ci
+  pr:
+    needs: install
+    strategy:
+      matrix:
+        node: [16.x, 18.x]
+        os: [macos-latest, ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+    env:
+      VITE_FR_AM_URL: ${{ secrets.VITE_FR_AM_URL }}
+      VITE_FR_AM_COOKIE_NAME: ${{ secrets.VITE_FR_AM_COOKIE_NAME }}
+      VITE_FR_OAUTH_PUBLIC_CLIENT: ${{ secrets.VITE_FR_OAUTH_PUBLIC_CLIENT }}
+      VITE_FR_REALM_PATH: ${{ secrets.VITE_FR_REALM_PATH }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Restore npm dependencies
+        uses: actions/cache@v3
+        id: npm-cache
+        with:
+          path: |
+            node_modules
+          key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-npm
       - run: npm run build
-        env:
-          VITE_FR_AM_URL: ${{ secrets.VITE_FR_AM_URL }}
-          VITE_FR_AM_COOKIE_NAME: ${{ secrets.VITE_FR_AM_COOKIE_NAME }}
-          VITE_FR_OAUTH_PUBLIC_CLIENT: ${{ secrets.VITE_FR_OAUTH_PUBLIC_CLIENT }}
-          VITE_FR_REALM_PATH: ${{ secrets.VITE_FR_REALM_PATH }}
-      - name: Publish to Chromatic
-        if: ${{ matrix.os == 'ubuntu-latest' && matrix.node < 17 && !github.event.pull_request.draft }}
-        uses: chromaui/action@v1
-        with:
-          projectToken: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
+      - name: Build Storybook
+        run: npm run build:storybook --quiet
+      - run: npx playwright install
+      - name: Serve Storybook and run tests
+        if: ${{ runner.os == 'ubuntu-latest' }} # this one hangs on mac os for some reason
+        run: |
+          npx concurrently -k -s first -n "SB,TEST" -c "magenta,blue" \
+            "npx http-server storybook-static --port 6006 --silent" \
+            "npx wait-on tcp:6006 && npm run test:storybook"
+  visual:
+    runs-on: ubuntu-latest
+    if: github.event_name != 'pull_request'
```

Comment:
so chromatic needs to run on push according to the docs. I believe this is because they require a commit histoyr that you get from the `push` event. 

This adds complexity because we can't just trigger it on a label or a draft state because those are pull_request event values (my understanding).

So - the `onlyChanged` argument should help alleviate some builds but - it won't be perfect.

Thoughts @daveadams56 @cerebrl ? I know this isn't ideal. But thank you for reminding me as i forgot this.

Reply:
Does it just break Chromatic entirely? What is the loss of functionality if we don't trigger it per push?

The code in it's current state, what does it do? Does it run Chromatic at every push still, or are we not running it at all ... ? Sorry for all the questions, but I feel a bit lost on this particular portion of our CI.

It's also worth noting that running the Storybook tests are not needed in CI if we run Chromatic. They should be mutually exclusive.

Reply:
We discussed this on slack but here is a sumary:

https://www.chromatic.com/docs/github-actions#recommended-configuration-for-build-events 

`push` means any push event (unless a push has more than 3 tags I think).

It does run it on every push. Its the recommended way. So i'm not 100% sure on what to do.

1. label -> requires the pull_request event
2. draft mode -> requires the pull_request event
3. push -> required for commit history and builds. Previously we were running on pull_request.

So our summary was lets see how it works on push, and see how builds are going. We should then after look to purchase it. In the event we are not allowed to purchase it, and builds are becoming a problem, we will look into another way of handling this. but as of now this is the documented suggested way.

Reply:
I'm ok with the explanation in chromatic docs, and then see if it causes a problem running on every push. Happy with your suggested way forward Ryan.